File: src/main/java/upbrella/be/umbrella/dto/response/UmbrellaResponse.java
package upbrella.be.umbrella.dto.response;

import lombok.Builder;
import lombok.Getter;
import upbrella.be.umbrella.entity.Umbrella;

@Getter
@Builder
public class UmbrellaResponse {

    private long id;
    private long storeMetaId;
    private String storeName;
    private long uuid;
    private boolean rentable;

    public static UmbrellaResponse fromUmbrella(Umbrella umbrella) {
        return UmbrellaResponse.builder()
                .id(umbrella.getId())
                .rentable(umbrella.isRentable())
                .storeMetaId(umbrella.getStoreMeta().getId())
                .storeName(umbrella.getStoreMeta().getName())
                .uuid(umbrella.getUuid())
                .build();
    }
}

File: src/main/kotlin/upbrella/be/umbrella/dto/response/UmbrellaResponse.kt
package upbrella.be.umbrella.dto.response;

data class UmbrellaResponse (

    val id: Long,
    val historyId: Long? = null,
    val storeMetaId: Long? = null,
    val storeName: String,
    val uuid: Long,
    val rentable: Boolean,
    val etc: String? = null
)
{
    companion object {
        fun fromUmbrella(umbrellaWithHistory: UmbrellaWithHistory): UmbrellaResponse {

            return UmbrellaResponse(
                id = umbrellaWithHistory.id,
                historyId = umbrellaWithHistory.historyId,
                rentable = umbrellaWithHistory.rentable,
                storeMetaId = umbrellaWithHistory.storeMeta.id,
                storeName = umbrellaWithHistory.storeMeta.name,
                uuid = umbrellaWithHistory.uuid,
                etc = umbrellaWithHistory.etc
            )
        }
    }
}
